scripts/scripts.js

```diff
@@ -819,165 +452,310 @@ export function addFavIcon(href) {
   }
 }
 
-async function addSegmentToIndex(url, index, pageSize) {
-  const resp = await fetch(url);
-  const json = await resp.json();
-  const complete = (json.limit + json.offset) === json.total;
-  json.data.forEach((post) => {
-    index.data.push(post);
-    index.byPath[post.path.split('.')[0]] = post;
-  });
-  index.complete = complete;
-  index.offset = json.offset + pageSize;
-}
-
 /**
- * fetches blog article index.
- * @returns {object} index with data and path lookup
+ * load LCP block and/or wait for LCP in default content.
  */
-
-export async function fetchBlogArticleIndex() {
-  const pageSize = 1000;
-  window.blogIndex = window.blogIndex || {
-    data: [],
-    byPath: {},
-    offset: 0,
-    complete: false,
-  };
-  if (window.blogIndex.complete) return (window.blogIndex);
-  const index = window.blogIndex;
-  const { offset } = index;
-  await addSegmentToIndex(`${getRootPath()}/query-index.json?limit=${pageSize}&offset=${offset}`, index, pageSize);
-  if (getRootPath() === '/uk/blog' || getRootPath() === '/au/blog') {
-    await addSegmentToIndex(`/blog/query-index.json?limit=${pageSize}&offset=${offset}`, index, pageSize);
-    index.data.sort((a, b) => b.date - a.date);
-  }
-  return (index);
-}
-
-export function makeLinkRelative(href) {
-  const url = new URL(href);
-  const host = url.hostname;
-  if (host.endsWith('.page') || host.endsWith('.live') || host === 'business.adobe.com') return (`${url.pathname}${url.search}${url.hash}`);
-  return (href);
-}
-
-export function rewritePath(path) {
-  let newpath = path;
-  const replacements = [{
-    from: 'news',
-    to: 'the-latest',
-  }, {
-    from: 'insights',
-    to: 'perspectives',
-  }];
-  replacements.forEach((r) => {
-    newpath = newpath.replace(`/${r.from}/`, `/${r.to}/`);
+async function waitForLCP() {
```

Comment:
that's odd. the `waitForLCP()` is definitely required in the `loadEager()`

Reply:
> that's odd. the `waitForLCP()` is definitely required in the `loadEager()`

I think their loadEager() is doing it similar thing itself.
<img width="900" alt="image" src="https://user-images.githubusercontent.com/55804872/159363782-ebc88464-95d4-4a52-af35-9a43f3605daa.png">

Reply:
I've updated to use waitForLCP() in loadEager() instead.

Reply:
i think you should remove that code and use `waitForLCP` instead, to be able to update things more easily going forward. there is no need to do this the old way, the `waitForLCP` code encapsulates this exact code in way that makes it easy to update.